src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/BrokeredServices/ServiceBrokerFactory.cs

```diff
@@ -64,6 +69,11 @@ public async Task CreateAsync()
         Contract.ThrowIfFalse(_container == null, "We should only create one container.");
 
         _container = await BrokeredServiceContainer.CreateAsync(_exportProvider, _cancellationTokenSource.Token);
+
+        foreach (var onInitialized in _onServiceBrokerInitialized)
+        {
+            onInitialized.OnServiceBrokerInitialized(_container.GetFullAccessServiceBroker());
```

Comment:
Are you concerned about exceptions being thrown from these methods?

Reply:
Good catch!. Because this call ends calling into a Service Workspace, it's probably a good idea to guard against exceptions.